src/main/java/duke/taskmanagement/TaskManager.java

```diff
@@ -0,0 +1,186 @@
+package duke.taskmanagement;
+
+import duke.customexceptions.EmptyDescriptionException;
+import duke.customexceptions.EmptyTimingDetailsException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class TaskManager {
+    private int taskUniqueID = 0;
+    private static ArrayList<Task> taskList = new ArrayList<>();
+    private static TaskRecorder taskRecorder = new TaskRecorder();
+
+    public TaskManager() {
+        try {
+            ArrayList<String[]> fileData = taskRecorder.loadData();
+            for (String[] data : fileData) {
+                String input = data[1];
+                String command = data[1].split(" ")[0];
+                String description = getDescription(input);
+                switch (command) {
+                case "todo":
+                    taskList.add(new Todo(description, taskUniqueID));
+                    break;
+                case "deadline":
+                    String by = getTimingDetails(input);
+                    taskList.add(new Deadline(description, taskUniqueID, by));
+                    break;
+                case "event":
+                    String at = getTimingDetails(input);
+                    taskList.add(new Event(description, taskUniqueID, at));
+                    break;
+                }
+                if (data[0].equals("1")) {
```

Comment:
How about [making the code more obvious](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-code-obvious) by using enumeration on the conditional?